express-crud/server.ts

```diff
@@ -0,0 +1,135 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+    select *
+    from "grades";
+    `;
+    const result = await db.query(sql);
+    res.json(result.rows[0]);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (!Number.isInteger(+gradeId)) {
+      throw new ClientError(400, 'a valid gradeId is required');
+    }
+    const sql = `
+    select *
+    from "grades"
+    where "gradeId" = $1;
+    `;
+    const result = await db.query(sql, [gradeId]);
+    const grade = result.rows[0];
+    if (!grade) {
+      throw new ClientError(404, 'must be a valid gradeId');
+    }
+    res.json(grade);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+    if (name === undefined || course === undefined || score === undefined) {
+      throw new ClientError(400, 'Name, Course, and Score are required');
+    }
+    if (score < 0 || score > 100 || !Number.isInteger(score)) {
```

Comment:
You should check that the number is an integer before checking its value because if it's not an integer, can't be between 0 and 100.